### PR TITLE
feat: InfluxQL learns how to plan some InfluxQL queries

### PR DESCRIPTION
**Pull request by stuartcarnie**
Helps #6112

## Summary

Teach the IOx querier how to transform a subset of an InfluxQL abstract syntax tree (AST) into a DataFusion `LogicalPlan`.

## InfluxQL Transformation

The most significant contribution of this PR is to the `plan::influxql` module in the `iox_query` crate. The `InfluxQLToLogicalPlan` data type has advanced sufficiently such that it is capable of translating a subset of the InfluxQL AST to a DataFusion `LogicalPlan` such that it can be executed. The features currently supported:

* The `SELECT` projection list is translated to a list of DataFusion `Expr` nodes
  * :white_check_mark: Arithmetic expressions
    * `5 + 5`, 
    * `field_1 + field_2`,
    * `field_1 * 5`
  * :white_check_mark: All supported InfluxQL scalar mathematical functions (`floor`, `ceil`, `atan2`, etc)
  * :white_check_mark: `*` wildcard expansion, including `::tag` and `::field` filters to select a subset of columns
  * :white_check_mark: `/<regex>/` regular expression expansion
  * :white_check_mark: column alias collision resolution
    * :exclamation: Important for clients which may rely on existing naming rules
  * :white_check_mark: column alias name resolution matches Go
    * :exclamation: Important for clients which may rely on existing naming rules
* The `FROM` clause allows a single measurement, that must exist
  * :white_check_mark: rewriter will expand regular expressions and validate measurement names exist in the schema
  * `InfluxQLToLogicalPlan` supports a single measurement
    * :white_check_mark: `FROM cpu`
    * multiple measurements not supported
      * :x: `FROM cpu, disk`
      * :x: `FROM /disk/` (matches `disk` and `diskio`)
    * Subqueries not supported
* `WHERE` clause 
  * :white_check_mark: `<`, `<=`, `==`, `!=`, `>=`, `>` conditional operators 
  * :x: `=~` and `!~` regular expression operators
  * :x: `now` function (some initial work has been done)
  * :x: Various timestamp literals
* :x: `GROUP BY`
* :x: `ORDER BY time`
* :white_check_mark: `LIMIT` and `OFFSET` clauses
* :white_check_mark: `SLIMIT` and `SOFFSET` clauses return a `Not Implemented` error
  * Why not implemented?
  * `SLIMIT` and `SOFFSET` don't work as expected per https://github.com/influxdata/influxdb/issues/7571
  * This issue [is noted](https://docs.influxdata.com/influxdb/v1.8/query_language/explore-data/#the-slimit-clause) in our official documentation
  * For the time being these will remain as such as current analysis of Cloud 2 InfluxQL queries reveals **no** usage of these clauses
* :x: `TZ` clause support
  * Remains a low priority, as 1,935 queries (0.069%) of 2,819,498 use the `TZ` clause

### CODE REVIEW

File: iox_query/src/plan/influxql.rs [hunk 1]
```diff
@@ -61,58 +122,583 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
...
+        //
+        assert_snapshot!(plan("SELECT f64_field FROM data"));
+        assert_snapshot!(plan("SELECT time, f64_field FROM data"));
+        assert_snapshot!(plan("SELECT time as timestamp, f64_field FROM data"));
+        assert_snapshot!(plan("SELECT foo, f64_field FROM data"));
+        assert_snapshot!(plan("SELECT foo, f64_field, i64_field FROM data"));
+        assert_snapshot!(plan("SELECT /^f/ FROM data"));
+        assert_snapshot!(plan("SELECT * FROM data"));
+
+        // The following is an outline of additional scenarios to develop
```
Comment (stuartcarnie) [inline iox_query/src/plan/influxql.rs:608]: As I think of new test scenarios and test cases, I will add them here and add / uncomment the individual cases as the implementation in this module becomes more advanced.

File: iox_query/src/test.rs [hunk 2]
```diff
@@ -177,11 +186,117 @@ impl ExecutionContextProvider for TestDatabase {
...
+            _ => None,
+        }
+    }
+}
+
+struct TestDatabaseSchemaProvider {
+    partitions: BTreeMap<String, BTreeMap<ChunkId, Arc<TestChunk>>>,
+}
+
+impl SchemaProvider for TestDatabaseSchemaProvider {
```
Comment (stuartcarnie) [inline iox_query/src/test.rs:235]: Currently this is required by the rewriter, but may changed to use `QueryNamespaceMeta` after @tustvold integrates their changes.

File: query_tests/src/influxql.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            output = Some(current);
+        }
+    }
+    output.expect("Expected at least one scenario")
+}
+
+#[tokio::test]
+async fn sql_select() {
+    use run_influxql_test_case as run;
+    assert_snapshot!(run(MultiChunkSchemaMerge {}, "SELECT host, \"user\" FROM cpu").await);
```
Comment (stuartcarnie) [inline query_tests/src/influxql.rs:94]: I will expand this as the InfluxQL AST → `LogicalPlan` transformation becomes more advanced.

**Q:** It is unclear to me when to use the `end_to_end_cases` vs the `query_test` crate for testing the query engine.

File: test_helpers_end_to_end/src/steps.rs [hunk 4]
```diff
@@ -125,10 +126,18 @@ pub enum Step {
...
+    InfluxQLQuery {
+        query: String,
+        expected: Vec<&'static str>,
+    },
+
     /// Run an InfluxQL query that's expected to fail using the FlightSQL interface and verify that the
     /// request returns the expected error code and message
     InfluxQLExpectingError {
-        sql: String,
+        query: String,
```
Comment (stuartcarnie) [inline test_helpers_end_to_end/src/steps.rs:140]: It isn't really SQL, so changed to a more appropriate field name

File: iox_query/src/plan/influxql/test_utils.rs [hunk 5]
```diff
@@ -67,61 +67,107 @@ impl TableProvider for EmptyTable {
...
+pub(crate) mod database {
+    use super::*;
+
+    /// Return a set of chunks that make up the test database.
+    ///
+    /// ## NOTE
+    /// The chunks returned by this function start numbering their
+    /// IDs from 1,000,000. A caller may wish to merge additional chunks
+    /// by using IDs less than 1,000,000.
+    pub(crate) fn chunks() -> Vec<Arc<TestChunk>> {
```
Comment (stuartcarnie) [inline iox_query/src/plan/influxql/test_utils.rs:80]: A set of tables with different schemas that will allow us to exercise various types of wildcard matching and schema merging as the InfluxQL AST → `LogicalPlan` mapper becomes more advanced.

File: influxdb_iox/tests/end_to_end_cases/influxql.rs [hunk 6]
```diff
@@ -21,10 +21,48 @@ async fn influxql_returns_error() {
...
+                        .into(),
+            },
+        ],
+    )
+    .run()
+    .await
+}
+
+#[tokio::test]
+async fn influxql_select_returns_results() {
```
Comment (stuartcarnie) [inline influxdb_iox/tests/end_to_end_cases/influxql.rs:37]: It is unclear to me when we should use this crate vs the `query_tests` crate, however, I wanted to ensure it was possible to add InfluxQL tests, hence a couple of cases added to this crate.

File: iox_query/src/plan/influxql.rs [hunk 7]
```diff
@@ -61,58 +122,587 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
...
+                ConditionalOperator::EqRegex => Ok(Operator::RegexMatch),
+                ConditionalOperator::NotEqRegex => Ok(Operator::RegexNotMatch),
+                ConditionalOperator::Lt => Ok(Operator::Lt),
+                ConditionalOperator::LtEq => Ok(Operator::LtEq),
+                ConditionalOperator::Gt => Ok(Operator::Gt),
+                ConditionalOperator::GtEq => Ok(Operator::GtEq),
+                ConditionalOperator::And => Ok(Operator::And),
+                ConditionalOperator::Or => Ok(Operator::Or),
+                // NOTE: This is not supported by InfluxQL SELECT expressions, so it is unexpected
+                ConditionalOperator::In => Err(DataFusionError::Internal(
```
Comment (alamb) [inline iox_query/src/plan/influxql.rs:313]: I think internal errors bubble up to Internal errors in grpc -- it might make more sense to use `DataFusionError::Plan` here instead

File: iox_query/src/plan/influxql.rs [hunk 8]
```diff
@@ -61,58 +122,587 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
...
+        } else {
+            match scope {
+                ExprScope::Projection => Err(DataFusionError::NotImplemented(
+                    "aggregate and selector functions in projection list".into(),
+                )),
+                ExprScope::Where => {
+                    if name.eq_ignore_ascii_case("now") {
+                        Err(DataFusionError::NotImplemented("now".into()))
+                    } else {
+                        Err(DataFusionError::External(
```
Comment (alamb) [inline iox_query/src/plan/influxql.rs:394]:
```suggestion
                        Err(DataFusionError::Plan(
```

File: query_tests/src/influxql.rs (hunk 3 above)
Comment (alamb) [reply query_tests/src/influxql.rs:94]: I left a comment above

**Review by alamb — COMMENTED** [review]
I didn't review this PR in detail @stuartcarnie  but I gave it a quick once over and I think it is looking very nice 👌  

I left some minor comments but nothing major

File: influxdb_iox/tests/end_to_end_cases/influxql.rs (hunk 6 above)
Comment (carols10cents) [reply influxdb_iox/tests/end_to_end_cases/influxql.rs:37]: My plan is to port all `query_tests` to use the same *functionality* as this crate does, with the `MiniCluster` and `StepTest`s, but I was not going to move all tests from `query_tests` to `influxdb_iox/tests/end_to_end_cases`. For now, I've moved them to `query_tests2` and they use some of the cases files in `query_tests`, but I'm happy to move them elsewhere if that's not what's desired!

File: iox_query/src/plan/snapshots/iox_query__plan__influxql__test__select_raw-2.snap [hunk 9]
```diff
@@ -0,0 +1,6 @@
+---
```
[the remark this answers is not in the document]
Comment (carols10cents) [reply iox_query/src/plan/snapshots/iox_query__plan__influxql__test__select_raw-2.snap:1]: I think it'd be great to let `insta` do some of the work that `query_tests` is doing now with the `cases` files, and I'll look into revising `query_tests2` to use `insta`!

File: query_tests/src/lib.rs [hunk 10]
```diff
@@ -9,6 +9,8 @@
 #[rustfmt::skip]
 mod cases;
```
Comment (carols10cents) [inline query_tests/src/lib.rs:10]: FWIW I don't think we have any conflicts with what we're doing, as I've only gotten to the tests in `cases` so far.

File: influxdb_iox/tests/end_to_end_cases/influxql.rs (hunk 6 above)
Comment (alamb) [reply influxdb_iox/tests/end_to_end_cases/influxql.rs:37]: I think that is a good plan

Comment (stuartcarnie) [reply influxdb_iox/tests/end_to_end_cases/influxql.rs:37]: Great! I implemented additional enumeration cases to run InfluxQL queries in the step-based tests. One in a prior PR for expecting an error and one in this PR for a good result.

File: iox_query/src/plan/influxql.rs [hunk 11]
```diff
@@ -5,31 +5,81 @@ mod rewriter;
...
 /// InfluxQL query planner
 #[allow(unused)]
 #[derive(Debug)]
 pub struct InfluxQLToLogicalPlan<'a> {
     ctx: &'a IOxSessionContext,
-    state: SessionState,
     database: Arc<dyn QueryNamespace>,
+    /// Defines the value of the `now()` function.
+    now: chrono::NaiveDateTime,
+    state: SessionState,
```
[the remark this answers is not in the document]
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:63]: I presume you were not referring to a system or user variable, as I looked for implementations of the `VarProvider` trait in DataFusion and IOx, only to find it in a test crate:

https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/src/test/variable.rs

Given that, are you referring to the `query_execution_start_time` field?

https://github.com/apache/arrow-datafusion/blob/71b9baecd0a3c881f96e9994d922f3c1b3d61854/datafusion/physical-expr/src/execution_props.rs#L32

In DataFusion 15, this value is initialised to `0` and then set when the `create_physical_plan` is executed, which happens after the `InfluxQLToLogicalPlan` transformation occurs.

In DataFusion 16, I have noted this value is set when the `SessionState` is snapshotted:

https://github.com/apache/arrow-datafusion/blob/71b9baecd0a3c881f96e9994d922f3c1b3d61854/datafusion/core/src/execution/context.rs#L1009

Given we are still on DF 15 and the value is `0`, it seems we can revisit this when DataFusion 16 is merged.

File: iox_query/src/plan/influxql.rs (hunk 7 above)
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:306]: I originally looked at the `Plan` error, and found that the documentation suggested it was for the physical planning phase:

```
/// This error happens whenever a plan is not valid. Examples include
/// impossible casts.
```

The `Internal` case is documented as:

```
// DataFusions has internal invariants that we are unable to ask the compiler to check for us.
// This error is raised when one of those invariants is not verified during execution.
```

In the case of this particular condition of the `match`, it should be impossible, as the `SELECT` statement does not parse the `IN` expression at all.

WDYT?

Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:306]: I see we're using it in `field_rewrite.rs` and other places, so I'll follow convention. Thanks!

File: iox_query/src/plan/influxql.rs (hunk 8 above)
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:387]: Makes sense to use `Plan` here, thanks! I see we're also using this case in `field_rewrite.rs` and others.

**Comment by stuartcarnie** [comment]
## TL;DR  

* I've removed the `query_tests` changes with the intent that I'll follow any conventions established by @carols10cents for the work she is doing 
* I have a couple of follow-up questions for @alamb:
    - [ ] https://github.com/influxdata/influxdb_iox/pull/6520#discussion_r1064255236
    - [ ] https://github.com/influxdata/influxdb_iox/pull/6520#discussion_r1064254289

File: iox_query/src/plan/influxql.rs (hunk 7 above)
Comment (alamb) [reply iox_query/src/plan/influxql.rs:306]: I think the idea is that Internal errors are bugs that are in DataFusion's code, not from user input. The documentation definitely looks less than ideal however

Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:306]: Ok cool – I'll use `Plan` here also, prefixing the text with "internal error" or similar so that it lets us know of some InfluxQL invariant that was violated

File: iox_query/src/plan/influxql.rs (hunk 11 above)
[the remark this answers is not in the document]
Comment (alamb) [reply iox_query/src/plan/influxql.rs:63]: >  Given that, are you referring to the query_execution_start_time field?

Yes that is what I was referring to 

I think it is fine to revisit -- I was trying to say that DataFusion already has knowledge of how to plan  `now()` and their special semantics (like they should be the same value any time they appear in the same query)

[the remark this answers is not in the document]
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:63]: Great.

I realise we have a `now` function in DataFusion SQL, so I'll look to see how that is implemented and derived

**Review by alamb — COMMENTED** [review]
I have this on my queue to review but I ran out of time today

File: iox_query/src/plan/influxql.rs (hunk 11 above)
[the remark this answers is not in the document]
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:63]: @alamb awesome:

https://github.com/apache/arrow-datafusion/blob/c5e2594e99b01c12d4f6903cb998a62a5479455c/datafusion/physical-expr/src/functions.rs#L415-L420

I'll drop that field completely, as it is a simple rewrite to the built-in `now` function 🥳

[the remark this answers is not in the document]
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:61]: Removed

File: iox_query/src/test.rs (hunk 2 above)
Comment (stuartcarnie) [reply iox_query/src/test.rs:235]: Turns out this cluster of traits on the `TestDatabase` is still needed by the planner to retrieve a table reference to generate logical plans.

File: iox_query/src/plan/influxql.rs (hunk 7 above)
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:297]: In this particular case, it should not be possible for a malformed query, as the `IN` conditional operator is not parsed for InfluxQL `SELECT` statements. 

This has been my distinction for using `Internal` in a few cases, where the rewriter or parser should never allow those conditions to occur once reaching the `InfluxQLToLogicalPlan` data type.

File: iox_query/src/plan/influxql.rs [hunk 12]
```diff
@@ -61,58 +104,583 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
...
+        let plan = self.plan_where_clause(select.condition, plan)?;
+
+        // Process and validate the field expressions in the SELECT projection list
+        let select_exprs = self.field_list_to_exprs(&plan, select.fields)?;
+
+        let projected_plan = self.project(plan.clone(), select_exprs.clone())?;
+        let mut combined_schema = (**projected_plan.schema()).clone();
+        combined_schema.merge(plan.schema());
+
+        let plan = project(plan, select_exprs)?;
```
Comment (alamb) [inline iox_query/src/plan/influxql.rs:133]: I don't understand this code -- it seems to be doing the projection two times (though ignoring the results of the first)

I wonder why isn't it more like

```suggestion
        // project with error checking
        let plan = self.project(plan, select_exprs)?;

```

File: iox_query/src/plan/influxql.rs [hunk 13]
```diff
@@ -61,58 +104,583 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
...
+        LogicalPlanBuilder::from(input).project(expr)?.build()
+    }
+
+    /// Validate the schema provides all of the columns referenced in the expressions.
+    fn validate_schema_satisfies_exprs(&self, schema: &DFSchema, exprs: &[Expr]) -> Result<()> {
+        find_column_exprs(exprs)
+            .iter()
+            .try_for_each(|col| match col {
+                Expr::Column(col) => match &col.relation {
+                    Some(_r) => {
```
Comment (alamb) [inline iox_query/src/plan/influxql.rs:193]: I don't quite follow why this logic is ignoring the relation (as in `Some(_r)` is ignored). For a column reference like `foo.bar` the relation (table) is `foo` but this code seems to just ignore the `foo`. Perhaps this isn't relevant until joins or subqueries (different relations) are more fully supported

Also, I recommend pulling the `Schema::try_from(arrow).map_err(|e| {` code to the top of this function rather than re-creating the IOx schema (which will recheck the embedded metadata) for each column reference.

File: iox_query/src/plan/influxql.rs [hunk 14]
```diff
@@ -61,58 +104,583 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
...
+            .map(|field| self.field_to_df_expr(field, plan))
+            .collect()
+    }
+
+    /// Map an InfluxQL [`Field`] to a DataFusion [`Expr`].
+    ///
+    /// A [`Field`] is analogous to a column in a SQL `SELECT` projection.
+    fn field_to_df_expr(&self, field: &Field, plan: &LogicalPlan) -> Result<Expr> {
+        let input_schema = plan.schema().as_ref().clone();
+        let expr = self.expr_to_df_expr(ExprScope::Projection, &field.expr, &input_schema)?;
```
Comment (alamb) [inline iox_query/src/plan/influxql.rs:249]: I think this is copying the entire schema (rather than an Arc). 

```suggestion
        let input_schema = plan.schema();
        let expr = self.expr_to_df_expr(ExprScope::Projection, &field.expr, &input_schema)?;
```

When poking around I also saw the same pattern in `plan_where_clause`

File: iox_query/src/plan/influxql.rs [hunk 15]
```diff
@@ -61,58 +104,583 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
...
+                // NOTE: This is not supported by InfluxQL SELECT expressions, so it is unexpected
+                ConditionalOperator::In => Err(DataFusionError::Internal(
+                    "unexpected binary operator: IN".into(),
+                )),
+            }?,
+            Box::new(self.conditional_to_df_expr(rhs, schema)?),
+        )))
+    }
+
+    /// Map an InfluxQL [`Expr`][`IQLExpr`] to a DataFusion [`Expr`].
```
Comment (alamb) [inline iox_query/src/plan/influxql.rs:305]:
```suggestion
    /// Map an InfluxQL [`IQLExpr`] to a DataFusion [`Expr`].
```

File: iox_query/src/plan/influxql.rs [hunk 16]
```diff
@@ -61,58 +104,583 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
...
+}
+
+/// Returns true if the field list contains a `time` column.
+///
+/// ⚠️ **NOTE**
+/// To match InfluxQL, the `time` column must not exist as part of a
+/// complex expression.
+fn has_time_column(fields: &FieldList) -> bool {
+    fields.iter().any(
+        |f| matches!(&f.expr, IQLExpr::VarRef { name, .. } if name.eq_ignore_ascii_case("time")),
```
Comment (alamb) [inline iox_query/src/plan/influxql.rs:496]: `normalize_identifier` says that columns are case sensitive but then this code is doing case insensitive matching. I am not sure if this is correct or not but it seems inconsistent

File: iox_query/src/plan/influxql.rs (hunk 1 above)
Comment (alamb) [reply iox_query/src/plan/influxql.rs:590]: Given this code added support for basic expressions, it might make sense to test a few things here like

```
        assert_snapshot!(plan("SELECT foo, f64_field  + 5 FROM data"));
        assert_snapshot!(plan("SELECT foo, 5.0 > f64_field FROM data"));
        assert_snapshot!(plan("SELECT foo, 5 > f64_field FROM data"));
        assert_snapshot!(plan("SELECT foo, sin(f64_field) FROM data"));

```

I apologize if this is not valid influxQL

File: iox_query/src/plan/influxql.rs [hunk 17]
```diff
@@ -61,58 +104,583 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
...
+    /// Tests to validate InfluxQL `SELECT` statements that project columns without specifying
+    /// aggregates or `GROUP BY time()` with gap filling.
+    #[test]
+    fn test_select_raw() {
+        //
+        // Selecting from a single measurement
+        //
+        assert_snapshot!(plan("SELECT f64_field FROM data"));
+        assert_snapshot!(plan("SELECT time, f64_field FROM data"));
+        assert_snapshot!(plan("SELECT time as timestamp, f64_field FROM data"));
```
Comment (alamb) [inline iox_query/src/plan/influxql.rs:584]: maybe also worth testing some case sensitivity -- like `time` and `TIME` and `f64_Field` etc

**Review by alamb — APPROVED** [review]
As always I found the code clear and easy to read and follow. I don't know InfluxQL well enough to be able to catch logical errors in this translation.

However,  the mechanics all make sense to me and I think maintaining this code will be straightforward. 

I had a few suggestions / comments but I don't think anything that would prevent this code from being merged

Nice work @stuartcarnie

File: iox_query/src/plan/influxql.rs (hunk 12 above)
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:133]: Ahh yes, good catch!

I had originally duplicated this from the `SqlToRel` type:

https://github.com/apache/arrow-datafusion/blob/9bee14ebd39dacbb66a9b1f34cd6494bc6a6be3f/datafusion/sql/src/planner.rs#L1056-L1059

And this code is used to allow the `GROUP BY` (and for SQL also the `HAVING`) clauses to use aliases from the projection. InfluxQL does not support this, so I can remove this redundant code completely.

File: iox_query/src/plan/influxql.rs (hunk 1 above)
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:544]: I will add some tests for arithmetic expressions, including functions, in the projection list 👍🏻  

For future reference, conditional expressions in the projection list aren't supported by InfluxQL (fails to parse)

File: iox_query/src/plan/influxql.rs (hunk 17 above)
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:538]: Good call.

That scenario is not working as is expected by InfluxQL, so I've established a pattern for tracking these issues as I go. This will allow me to track the problems with tests that fail, but are ignored until they are resolved. For example:

https://github.com/influxdata/influxdb_iox/blob/243ae650f9ddab8bf8f08f948c0ef7684da47b3c/iox_query/src/plan/influxql.rs#L679-L694

**Comment by stuartcarnie** [comment]
Thanks again for the review and feedback, @alamb! I've responded to your remaining feedback items, removed the redundant code and added additional tests.

File: iox_query/src/plan/influxql.rs (hunk 16 above)
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs:450]: Correct! This should only match `time`, and I've added a test that includes a `TIME` field to validate the distinction.

File: iox_query/src/plan/influxql.rs (hunk 13 above)
Comment (stuartcarnie) [reply iox_query/src/plan/influxql.rs]: I've removed this entire block for now, as it was code for future use and to validate I could extract the IOx metadata (to identify fields vs tags) from the input schema.
